hedera-mirror-importer/src/test/java/com/hedera/mirror/importer/parser/record/ethereum/Eip2930EthereumTransactionParserTest.java

```diff
@@ -0,0 +1,107 @@
+import com.hedera.mirror.importer.exception.InvalidEthereumBytesException;
+import java.math.BigInteger;
+import lombok.SneakyThrows;
+import org.apache.commons.codec.binary.Hex;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+class Eip2930EthereumTransactionParserTest extends AbstractEthereumTransactionParserTest {
+    public static final String EIP_2930_RAW_TX =
+            "01f87182012a8085a54f4c3c00832dc6c094000000000000000000000000000000000000052d8502540be40083123456c001a04d83230d6c19076fa42ef92f88d2cb0ae917b58640cc86f221a2e15b1736714fa03a4643759236b06b6abb31713ad694ab3b7ac5760f183c46f448260b08252b58";
+
+    @BeforeAll
+    static void beforeAll() {
+        ethereumTransactionParser = new Eip2930EthereumTransactionParser();
+    }
+
+    @SneakyThrows
+    @Override
+    public byte[] getTransactionBytes() {
+        return Hex.decodeHex(EIP_2930_RAW_TX);
+    }
+
+    @SneakyThrows
+    @Test
+    void decodeEip2930Transaction() {
+        var ethereumTransaction = ethereumTransactionParser.decode(Hex.decodeHex(EIP_2930_RAW_TX));
+        validateEthereumTransaction(ethereumTransaction);
+        assertThat(ethereumTransaction.getType()).isEqualTo(1);
+    }
+
+    @Test
+    void decodeEip1559Type() {
+        var ethereumTransactionBytes = RLPEncoder.sequence(Integers.toBytes(2), new Object[] {});
+
+        assertThatThrownBy(() -> ethereumTransactionParser.decode(ethereumTransactionBytes))
+                .isInstanceOf(InvalidEthereumBytesException.class)
+                .hasMessage("Unable to decode EIP2930 ethereum transaction bytes, 1st byte was 2 but should be 1");
+    }
+
+    @Test
+    void decodeNonListRlpItem() {
+        var ethereumTransactionBytes = RLPEncoder.sequence(Integers.toBytes(1), Integers.toBytes(1));
+
+        assertThatThrownBy(() -> ethereumTransactionParser.decode(ethereumTransactionBytes))
+                .isInstanceOf(InvalidEthereumBytesException.class)
+                .hasMessage("Unable to decode EIP2930 ethereum transaction bytes, 2nd RLPItem was not a list");
+    }
+
+    @Test
+    void decodeIncorrectRlpItemListSize() {
+        var ethereumTransactionBytes = RLPEncoder.sequence(Integers.toBytes(1), new Object[] {});
+
+        assertThatThrownBy(() -> ethereumTransactionParser.decode(ethereumTransactionBytes))
+                .isInstanceOf(InvalidEthereumBytesException.class)
+                .hasMessage("Unable to decode EIP2930 ethereum transaction bytes, 2nd RLPItem list size was 0 but "
+                        + "should be 11");
+    }
+
+    @Override
+    protected void validateEthereumTransaction(EthereumTransaction ethereumTransaction) {
+        assertThat(ethereumTransaction)
+                .isNotNull()
+                .satisfies(t -> assertThat(t.getType()).isEqualTo(Eip2930EthereumTransactionParser.EIP2930_TYPE_BYTE))
+                .satisfies(t -> assertThat(Hex.encodeHexString(t.getChainId())).isEqualTo("012a"))
```

Comment:
Most if not all can be converted to the simpler returns:

```suggestion
                .returns(EIP2930_TYPE_BYTE, EthereumTransaction::getType)
                .returns("012a", EthereumTransaction::getChainId)
```